### PR TITLE
[New Integration] Google Cloud Pub/Sub

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/23146

## Description
New Integration `Google Cloud Pub/Sub`.

## Screenshots
image

## Minimum version of Demisto
- [x] 5.0.0

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/GooglePubSub/Integrations/GooglePubSub/GooglePubSub_test.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+        """
+        Given:
+            - project_id
+            - topic_id
+            - subscription_id
+        When:
+            - we want topic subscription_name
+        Then:
+            - GoogleNameParser should parse it in the expected format
+        """
```
Comment (yuvalbenshalom) [inline Packs/GooglePubSub/Integrations/GooglePubSub/GooglePubSub_test.py:66]: good tests docs 👍

File: Packs/GooglePubSub/Integrations/GooglePubSub/README.md [hunk 2]
```diff
@@ -0,0 +1,958 @@
...
+                },
+                "pushEndpoint": "https://d4d1290519676f29baf13a7bf18a25bf-dot-j40fd5d18d8c290e1p-tp.appspot.com/_ah/push-handlers/pubsub/projects/dmst-integrations/topics/dmst-topic?pubsub_trigger=true"
+            },
+            "topic": "projects/dmst-integrations/topics/dmst-topic"
+        }
+    ]
+}
+```
+
+##### Human Readable Output
```
Comment (yuvalbenshalom) [inline Packs/GooglePubSub/Integrations/GooglePubSub/README.md:399]: not sure why this part is not marked blue. is that a GitHub issue?

File: Packs/GooglePubSub/Integrations/GooglePubSub/GooglePubSub.py [hunk 3]
```diff
@@ -0,0 +1,1164 @@
...
+
+    @staticmethod
+    def get_snapshot_project_name(project_id, snapshot_id):
+        return GoogleNameParser.get_project_name(
+            project_id
+        ) + GoogleNameParser.FULL_SNAPSHOT_PREFIX.format(snapshot_id)
+
+
+# disable-secrets-detection-start
+class BaseGoogleClient:
```
[the remark this answers is not in the document]
Comment (DeanArbel) [reply Packs/GooglePubSub/Integrations/GooglePubSub/GooglePubSub.py:67]: So it'll be easier to use the same resources as @orlichter1 on his PR, and so it'll be easier to move later the base `GoogleClient` to another package - i.e. `BasePackage`

File: Packs/GooglePubSub/Integrations/GooglePubSub/GooglePubSub.py [hunk 4]
```diff
@@ -0,0 +1,1164 @@
...
+                .subscriptions()
+                .pull(subscription=sub_name, body=req_body)
+                .execute()
+        )
+
+    def ack_messages(self, sub_name, acks):
+        """
+        Ack a list of messages
+        :param sub_name: subscription name
+        :param acks: ack ids to ack
```
[the remark this answers is not in the document]
Comment (DeanArbel) [reply Packs/GooglePubSub/Integrations/GooglePubSub/GooglePubSub.py:236]: These are actually ack ids - i.e. ids uses specifically to ack a message. They are different ids than the message ids

File: Packs/GooglePubSub/Integrations/GooglePubSub/README.md (hunk 2 above)
Comment (DeanArbel) [reply Packs/GooglePubSub/Integrations/GooglePubSub/README.md:399]: probably. The MD looks good inhttps://github.com/demisto/content/blob/8e3c54debe4eaf77526170709caf6c61a43b131c/Packs/GooglePubSub/Integrations/GooglePubSub/README.md

**Comment by DeanArbel** [comment]
@yuvalbenshalom here is an example list of errors user can encounter:

##Configuration errors:
### Incorrect `Service Account Private Key`
image

### Incorrect `Subscription`
image

## Errors for incorrect artifacts (i.e. incorrect id given for `project`, `subscription`, `topic` or `snapshot`):
image

## Errors for incorrect datetime format in `gcp-pubsub-topic-messages-seek`
image

## Error for incorrect page_token
image

**Comment by DeanArbel** [comment]
@kirbles19 Please review the following the files:
* [CHANGELOG.md](https://github.com/demisto/content/blob/a9091ab8f6608a282f52374210358fbf31d55572/Packs/GooglePubSub/Integrations/GooglePubSub/CHANGELOG.md)
* [yml file](https://github.com/demisto/content/blob/a9091ab8f6608a282f52374210358fbf31d55572/Packs/GooglePubSub/Integrations/GooglePubSub/GooglePubSub.yml)
* (detailed description)[https://github.com/demisto/content/blob/a9091ab8f6608a282f52374210358fbf31d55572/Packs/GooglePubSub/Integrations/GooglePubSub/GooglePubSub_description.md]
* (readme)[https://github.com/demisto/content/blob/a9091ab8f6608a282f52374210358fbf31d55572/Packs/GooglePubSub/Integrations/GooglePubSub/README.md]

**Comment by kirbles19** [comment]
@DeanArbel Done.

**Pull request merged**